.github/workflows/worker-playground-preview-deploy-testing.yml

```diff
@@ -1,58 +1,66 @@
-name: Playground Worker tests
+name: Deploy Playground Preview Worker (testing)
 
+# On a push to `main`, on Cloudflare, where there are changes to the files in this worker's package,
+# or an update to a PR, on Cloudflare, labelled as `playground-worker`,
+# deploy to testing and then run the end-to-end tests against this deployment.
 on:
   push:
     branches:
       - main
-      - changeset-release/main
+    paths:
+      - "packages/playground-preview-worker/**"
+
   pull_request:
     types: [synchronize, opened, reopened, labeled, unlabeled]
-  repository_dispatch:
 
 jobs:
   e2e-test:
-    if: github.repository_owner == 'cloudflare' && (github.event_name != 'pull_request' || (github.event_name == 'pull_request' && contains(github.event.*.labels.*.name, 'playground-worker' )) || (github.event_name == 'pull_request' && github.head_ref == 'changeset-release/main'))
-    name: "Playground Worker Test"
+    if: github.repository_owner == 'cloudflare' && (github.event_name != 'pull_request' || contains(github.event.*.labels.*.name, 'playground-worker'))
```

Comment:
The triggering based on a label should probably be removed in favour of the path-based triggering above

Reply:
Yeah I was wondering about that. Here is where I landed:

Since the playground preview tests all share the same deployed testing environment Worker I think it is safer to restrict this to labeled PRs to avoid collisions - it always runs on pushes to `main` anyway.